### PR TITLE
feat: adds restrictive content security policies

### LINKED ISSUE #2047: ️🛠 - [Task tracker] Improve Content Security Policies
### Task description

Fix issue reported here: [[Transak] Missing Content Security Policy Directives in Bloom Application](https://github.com/AuditOneAuditReviews/bloom_sc_audit_review/issues/22)

We need to define more strict content security policies for our app to make it more secure, this is the list of policies that we can add/improve:

- [x] #2144 
- [x] `default-src`
- [x] `connect-src`
- [x] `img-src`
- [x] `base-uri`
- [x] `form-action`
- [x] `frame-src`
- [ ] ~`frame-ancestors`~ Should be in request headers, not in `<meta>`
- [x] `worker-src`
- [x] `script-src-attr`
- [x] `style-src-elem`
- [x] `style-src-attr`
- [x] `object-src`
- [x] `media-src`
- [x] `font-src`
- [x] `manifest-src`
- [ ] ~`prefetch-src`~ Unrecognized directive
- [x] `navigate-to`
- [ ] ~`report-to`~ Needs to add Sentry endpoint
- [ ] ~`sandbox`~ Should be in request headers, not in `<meta>`
- [x] `upgrade-insecure-requests`

### Requirements

_No response_

### PR DESCRIPTION
**Pull request by jeeanribeiro**
## Summary

closes #2047 

### Platforms

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [x] Windows

### Instructions

- Test features that could be affected by security policies (third party requests)

## Checklist

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [x] I have verified that new and existing unit tests pass locally with my changes
-   [x] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/public/about.html [hunk 1]
```diff
@@ -6,6 +6,28 @@
...
+          default-src 'self';
+          connect-src 'self';
+          img-src 'self';
+          base-uri 'self';
+          form-action 'none';
+          frame-src 'none';
+          worker-src 'none';
+          script-src-elem 'self' 'unsafe-inline';
+          script-src-attr 'none';
+          style-src-elem 'self' 'unsafe-inline';
```
Comment (Tuditi) [inline packages/desktop/public/about.html]: Why do we need `unsafe-inline` inside the about window?

File: packages/desktop/public/error.html [hunk 2]
```diff
@@ -5,6 +5,27 @@
...
+        content="
+            default-src 'self';
+            connect-src 'self';
+            img-src 'self';
+            base-uri 'self';
+            form-action 'none';
+            frame-src 'none';
+            worker-src 'none';
+            script-src 'self' 'unsafe-inline';
+            style-src-elem 'self' 'unsafe-inline';
```
Comment (Tuditi) [inline packages/desktop/public/error.html]: Same here. Why do we need unsafe-inline for the error tab?

**Review by Tuditi — COMMENTED** [review]
I think we could improve security be extracting the css and the scripts into separate files. This might also reduce code duplication inside the styles. What do you think?

File: packages/desktop/public/about.html (hunk 1 above)
Comment (jeeanribeiro) [reply packages/desktop/public/about.html]: Refactored style and script to be in a separate file

File: packages/desktop/public/error.html (hunk 2 above)
Comment (jeeanribeiro) [reply packages/desktop/public/error.html]: Refactored style and script to be in a separate file

**Review by Tuditi — APPROVED** [review]
LGTM!

**Pull request merged**